docs/troubleshooting.md

```diff
@@ -62,3 +64,63 @@ For instance, try this:
 # oc create --raw /apis/admission.hive.openshift.io/v1/dnszones -f config/samples/hiveadmission-review-failure.json -v 8 | jq
 ```
 
+## Regaining access to a cluster after changing the API certificate
+
+There are several ways to regain access to a cluster after its API certificate has been changed.  Usually this will manifest as an error message similar to: `x509: certificate signed by unknown authority`.
+
+### Replacing the API certificate of a managed single cluster with kubeconfig
+
+In order for a new certificate for a cluster to be accepted by the system, it must be added to the kubeconfig, alongside or replacing any existing certificates. Start by inspecting your current kubeconfig:
+
+```sh
+$ export MY_KUBECONFIG_NAME=$(oc get cd $CLUSTERNAME -o "jsonpath={.spec.clusterMetadata.adminKubeconfigSecretRef.name}")
+$ oc get secret $MY_KUBECONFIG_NAME -o "jsonpath={.data.kubeconfig}" | base64 --decode > my-kubeconfig-file
+$ cat my-kubeconfig-file
+```
+
+```yaml
+...
+clusters:
+- cluster:
+    certificate-authority-data: <base64 encoded blob>
+    server: https://api.<your cluster>.<your domain>:6443
+  name: <your cluster>
+...
+```
+
+The field `certificate-authority-data` is what needs to be updated. When base64 decoded, it looks something like this:
+
+```sh
+$ echo <base64 encoded blob> | base64 --decode > decoded-existing-certs.pem
+$ cat decoded-existing-certs.pem
+```
+
+```
+-----BEGIN CERTIFICATE-----
+<base64 encoded certificate blob>
+-----END CERTIFICATE-----
+...
+-----BEGIN CERTIFICATE-----
+<potentially more base64 encoded certificate blobs>
+-----END CERTIFICATE-----
+```
+
+Concatenate your new certificate with the existing decoded certificates:
+
+```sh
+$ cat decoded-existing-certs.pem <your new certificate full chain>.pem | openssl base64 -A > encoded-new-cert-bundle.pem
+```
+
+Copy the contents of this new base64 encoded blob and use it to replace the `certificate-authority-data` field in your kubeconfig. After the kubeconfig file is ready, patch your cluster to use the new kubeconfig:
+
+```sh
+oc patch secret $MY_KUBECONFIG_NAME --type='json' -p="[{'op': 'replace', 'path': '/data/kubeconfig', 'value': '$(openssl base64 -A -in my-updated-kubeconfig-file)'}]"
+```
+
+### Replacing the API certificate globally on the managing cluster with hiveconfig
+
+Hive can use its own configuration file to regain access to the managed cluster. Use the `hack/set-additional-ca.sh` script like so: 
+
+```sh
+$ hack/set-additional-ca.sh <file location of your chain.pem>
```

Comment:
unclear whether this is supposed to be the encoded (base64) or decoded (PEM) file.